lib/engine_socketio.js

```diff
@@ -118,10 +118,10 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
     return engineUtil.createLoopWithCount(requestSpec.count || -1, steps);
   }
 
-  let f = function(context, callback) {
+  let f = function(context, callback, socketio) {
```

Comment:
Is there a particular reason we are passing `socketio` as an extra argument here, rather than keeping it attached to `context`? That function's signature should really follow that of `async#waterfall` [task functions](https://github.com/caolan/async/blob/v1.5.2/README.md#waterfall) since that's where `f` is used. This would avoid us needing to wrap `f` in `preStep` as well.